core/src/main/java/com/linecorp/armeria/client/encoding/DecodingClient.java

```diff
@@ -124,7 +126,7 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
             return unwrap().execute(ctx, req);
         }
 
-        final List<String> encodings = ImmutableList.copyOf(ENCODING_SPLITTER.split(acceptEncoding));
+        final Set<String> encodings = ImmutableSet.copyOf(ENCODING_SPLITTER.split(acceptEncoding));
```

Comment:
optional; not directly related to this PR since `ImmutableSet.copyOf` preserves the original order, but what do you think of adding a unit test to prevent future regressions?

e.g. if `Accept-Encoding: gzip, deflate, br`, the order should be preserved.

Reply:
I'm not sure I understand the motivation of preserving the order in the set. IIUC the set is a temporary collection used to create the map of encoding->decoder factory, which in turn is used when decoding the response (by fetching the correct decoder according to the received encoding of the response).

```
final Set<String> encodings = ImmutableSet.copyOf(ENCODING_SPLITTER.split(acceptEncoding));
final ImmutableMap.Builder<String, StreamDecoderFactory> factoryBuilder =
        ImmutableMap.builderWithExpectedSize(encodings.size());

for (String encoding : encodings) { // only iterated here, and the order doesn't matter AFAIK
      final StreamDecoderFactory factory = decoderFactories.get(encoding);
      if (factory != null) {
          factoryBuilder.put(factory.encodingHeaderValue(), factory);
      }
  }
```
Do you mean to check if the order of the values in the header is preserved?

Reply:
Ah sorry, I think you're right 😅 

I was looking at the following line, and I thought `encodings` was being used for some reason.
https://github.com/line/armeria/blob/43bd08248ba86ab5942f781da33b11a42fe3b1a5/core/src/main/java/com/linecorp/armeria/client/encoding/DecodingClient.java#L147

Ignore my comment please 🙏